angular-workspace/projects/ni/nimble-angular/src/directives/patterns/link/nimble-link-base.ts

```diff
@@ -0,0 +1,66 @@
+import { ElementRef, Injectable, Input, Renderer2 } from '@angular/core';
+import type { AnchorBase } from '@ni/nimble-components/dist/esm/anchor-base';
+
+/**
+ * Base class for directives providing link attributes.
+ */
+@Injectable()
+export class NimbleLinkBase<T extends AnchorBase> {
```

Comment:
I don't think this class needs to be generic as long as the other anchor directives extend this class

Reply:
Without that generic type constraint, I get errors trying to reference the properties on `nativeElement` in the getters, e.g. `this.elementRef.nativeElement.href`. Also, isn't it just better to enforce that everything extending this derives from AnchorBase?

Reply:
makes sense that it's needed for the nativeElement refs 👍